### PR TITLE
Fixed QuerystringRegistryReader.mapSortableIndex so it doesnt return …

### LINKED ISSUE #37: CatalogError: The index chosen for sort_on (byline) is not capable of being used as a sort index.
The field 'Sort on'  in collection, list index of type 'ZCTextIndex' if the record is placed as 'sortable'. When you create a filter and an index 'ZCTextIndex' is selected in 'Sort on', the error below occurs  in the return of the ajax call (querybuilder_html_results):

    Module ZPublisher.Publish, line 138, in publish
    Module ZPublisher.mapply, line 77, in mapply
    Module ZPublisher.Publish, line 48, in call_object
    Module plone.app.querystring.querybuilder, line 58, in html_results
    Module plone.app.querystring.querybuilder, line 49, in __call__
    Module plone.app.querystring.querybuilder, line 116, in _makequery
    Module Products.CMFPlone.CatalogTool, line 393, in searchResults
    Module Products.ZCatalog.ZCatalog, line 604, in searchResults 
    Module Products.ZCatalog.Catalog, line 916, in searchResults
    Module Products.ZCatalog.Catalog, line 894, in _getSortIndex

    CatalogError: The index chosen for sort_on (byline) is not capable of being used as a sort index.

The field 'Sort on' should not list index of type 'ZCTextIndex'.

### PR DESCRIPTION
**Pull request by winstonf88**
ZCTextIndex type indexes are not sortable, QuerystringRegistryReader.mapSortableIndex should not return these type of indexes.
http://docs.plone.org/develop/plone/searching_and_indexing/query.html#available-indexes

closes #37

### CODE REVIEW

**Comment by idgserpro** [comment]
@winstonf88 I think you have to make this change in the 1.2.x branch also, which is the version used by the Plone 4.3.

**Comment by jensens** [comment]
Plone Contributor Agreement is missing, at least I cant find @winstonf88 in @plone

**Comment by idgserpro** [comment]
@hvelarde you can help here? We need it in https://github.com/plonegovbr/brasil.gov.portal/issues/191

**Comment by hvelarde** [comment]
@jensens he signed the agreement in 2013 during the Plone Conference; if the agreement is missing is our fault; do you want another agreement to be signed?

**Comment by jensens** [comment]
agreements are not my topic, i just check the list before merge. 
best send an email to agreements@plone.org and ask what happend.

**Comment by jensens** [comment]
and while checking it, please add a changelog entry.

**Comment by hvelarde** [comment]
@winstonf88 please download, sign, scan and send again the agreement to the address above; let us know when you're done with that.

**Comment by jensens** [comment]
meanwhile lets run tests http://jenkins.plone.org/job/pull-request-5.0/279/

**Comment by winstonf88** [comment]
@hvelarde @jensens Contributors agreement sent. Please check.

**Comment by ableeb** [comment]
@winstonf88 your Contributors agreement has been received and will be processed. Thanks!

**Comment by hvelarde** [comment]
@jensens @esteele seems that branch 1.2.x is missing from the repo and is the one used on Plone 4.3.

what can we do?

**Comment by idgserpro** [comment]
@winstonf88  it seems that the test failed: http://jenkins.plone.org/job/pull-request-5.0/279/
You can fix?

**Comment by winstonf88** [comment]
@idgserpro I have tested this on a developmento env and it works just fine. I think this error has something to do with the test environment, I don't really know how to fix this test. Any ideas on another way I can get portal_catalog ?

**Comment by hvelarde** [comment]
@winstonf88 you're missing a line on the changelog on all PR; I have no idea why the test is failing but we can focus on the 1.2.x branch first as that is the one we care for now, do you agree?

**Comment by idgserpro** [comment]
@hvelarde  I agree that the focus should be on the 1.2.x branch. @winstonf88  must create a new branch based on branch 1.2.x and create a new PR.

**Comment by idgserpro** [comment]
@winstonf88  I think the tests are failing because [the way it runs](https://github.com/plone/plone.app.querystring/blob/master/plone/app/querystring/tests/testRegistryReader.py#L128-L135), the [context](https://github.com/plone/plone.app.querystring/blob/6ef03795883b0a92aee85eb1b4f5e0a38e4c401e/plone/app/querystring/registryreader.py#L109) is not an object that supports acquisition. There are two ways to fix this:

* Fix the tests, changing the context of [reader object](https://github.com/plone/plone.app.querystring/blob/master/plone/app/querystring/tests/testRegistryReader.py#L133) for a collection (that's what happens when we use the functionality via browser).

* Using ```zope.app.component.hooks.getSite``` to get the portal and use in his function.

It will need fix this in other branchs too.

**Comment by hvelarde** [comment]
@idgserpro @winstonf88 don't import from `zope.app.component` but `zope.component`

**Comment by idgserpro** [comment]
@hvelarde I updated the comment. Thank you!

File: plone/app/querystring/registryreader.py [hunk 1]
```diff
@@ -6,8 +6,13 @@
 from zope.i18nmessageid import Message
 from zope.interface import implements
 from zope.schema.interfaces import IVocabularyFactory
+from zope.component.hooks import getSite
 import logging
 
+from Products.ZCTextIndex.interfaces import IZCTextIndex
+from Products.CMFCore.utils import getToolByName
```
Comment (idgserpro) [inline plone/app/querystring/registryreader.py]: It is good to keep the imports ordered.

**Comment by jensens** [comment]
http://jenkins.plone.org/job/pull-request-5.0/339/

**Comment by idgserpro** [comment]
@winstonf88  don't forget to fix the order of imports in other branchs also:

https://github.com/plone/plone.app.querystring/pull/50
https://github.com/plone/plone.app.querystring/pull/49

**Comment by hvelarde** [comment]
@jensens Jenkins is green

**Comment by jaroel** [comment]
https://github.com/plone/plone.app.querystring/commit/07edaa26dc7a76a96e7d3b6ec8ba20b75044d192 removed some commits. @ebrehault 's work is missing at least.

**Comment by esteele** [comment]
If someone can fix the merge in some way, I can cut a new release. Otherwise I'm going to hold off on including this in rc3.

File: plone/app/querystring/registryreader.py [hunk 2]
```diff
@@ -68,24 +70,19 @@ def getVocabularyValues(self, values):
         """Get all vocabulary values if a vocabulary is defined"""
 
         for field in values.get(self.prefix + '.field').values():
-            field['values'] = OrderedDict()
+            field['values'] = {}
```
Comment (jensens) [inline plone/app/querystring/registryreader.py]: This must be an OrderedDict, right?

**Comment by jensens** [comment]
i try to fix it. give me 20 mins ;)

**Pull request merged**